recipes/cppcheck/all/conanfile.py

```diff
@@ -16,6 +16,7 @@ class CppcheckConan(ConanFile):
     options = {"with_z3": [True, False], "have_rules": [True, False]}
     default_options = {"with_z3": True, "have_rules": True}
     exports_sources = ["CMakeLists.txt", "patches/**"]
+    build_policy = "always"
```

Comment:
We do not need this in CCI since we are creating the packages for consumers here

```suggestion
```

Reply:
> It looks the cfg files are moved into the bin folder, what error did you expereience?

If you run cppcheck on a directory containing C++ files
```
cppcheck --xml --enable=warning,style,performance --std=c++11 . 2> test.xml
cppcheck: Failed to load library configuration file 'std.cfg'. File not found
```

cppcheck doesn't seem to be relocatable. It uses a hard-coded absolute path to find the cfg files.
By default it looks in `<prefix>/share`, so removing the `share` directory as currently in the recipe is not a good idea.
I don't understand why they are manually copied to `bin/cfg/` in the first place, but didn't want to break anything.

The change in the `build_policy` is for the same reason. Since cppcheck is not relocatable, it needs to be build by every user locally to make sure that the hard-coded paths fit with the local machine.

Reply:
The "share" folder is unknown no to CCI recipes and most often is not required for consuming packages...

https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#kb-h013-default-package-layout

We could add an exception https://github.com/conan-io/hooks/blob/addcdacd875d18c3407f92ab2e7f4be58febf53e/hooks/conan-center.py#L780

***

Looking  at the code... I found https://github.com/danmar/cppcheck/blob/5768c3a726f524dd6a0d22b7f7aab8c5895794e3/cli/cppcheckexecutor.cpp#L854
Which starts to look in the current directory... maybe we should just drop it in the `bin` folder? 🤔

Reply:
Thanks. Didn't know that the *shared* directory is not preferred.

Reply:
Is the non-relocalibility of cppcheck caused by its use of [`FILESDIR`](https://github.com/danmar/cppcheck/blob/7f358b2bed112a0ef1a4bb116b23648977ff7907/lib/platform.cpp#L170-L176)?
We fixed a similar problem in swig by using relative paths to the executable.
Alternatively, one can introduce a `CPPCHECK_FILESDIR` environment variable.